src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -4027,26 +4040,21 @@ private DynamicMetaObject CannotIndexTarget(DynamicMetaObject target, DynamicMet
                 bindingRestrictions);
         }
 
-        internal static bool CanIndexFromEndWithNegativeIndex(DynamicMetaObject target)
+        internal static bool CanIndexFromEndWithNegativeIndex(
+            DynamicMetaObject target,
+            MethodInfo indexer,
+            ParameterInfo[] getterParams)
         {
-            var limitType = target.LimitType;
-            if (limitType.IsArray || limitType == typeof(string) || limitType == typeof(StringBuilder))
-            {
-                return true;
-            }
-
-            if (typeof(IList).IsAssignableFrom(limitType))
-            {
-                return true;
-            }
-
-            if (typeof(OrderedDictionary).IsAssignableFrom(limitType))
+            if (getterParams.Length != 1 || getterParams[0].ParameterType != typeof(int))
```

Comment:
Can getterParams be null?

Reply:
No, reflection will return empty arrays but not null.